Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -167,6 +183,25 @@ public class HTTPNetworkTransport {
     
     return task
   }
+
+  private func handleGraphQLErrorsOrComplete<Operation>(operation: Operation,
+                                                        response: GraphQLResponse<Operation>,
+                                                        completionHandler: @escaping (_ result: Result<GraphQLResponse<Operation>, Error>) -> Void) throws {
+    if let delegate = self.delegate as? NetworkGraphQLErrorDelegate,
```

Comment:
Personally I'd `guard let` this for clearer control flow.  

You may also want to look at some of the changes proposed in #767 - there's some stuff in there that's similar to this just without the delegate, including something to only parse the errors (rather than parsing the entire data structure as you're doing below).

Reply:
Added `guard let`

Related to the #767 are you thinking of the `func parseErrorsFast()`?

Reply:
yep!